src/trackers/impression.js

```diff
@@ -18,38 +18,46 @@ import logFactory from '../utils/logger';
 import thenable from '../utils/promise/thenable';
 const log = logFactory('splitio-client:impression-tracker');
 
-function ImpressionsTrackerContext(context) {
+function ImpressionTracker(context) {
   const collector = context.get(context.constants.STORAGE).impressions;
   const settings = context.get(context.constants.SETTINGS);
   const listener = settings.impressionListener;
+  const integrationsManager = context.get(context.constants.INTEGRATIONS_MANAGER, true);
   const { ip, hostname } = settings.runtime;
   const sdkLanguageVersion = settings.version;
 
   return {
-    track: function(impression, attributes) {
+    track: function (impression, attributes) {
       const res = collector.track([impression]);
 
       // If we're on an async storage, handle error and log it.
       if (thenable(res)) res.catch(err => {
         log.error(`Could not store impression. Error: ${err}`);
       });
 
-      // Wrap in a timeout because we don't want it to be blocking.
-      listener && setTimeout(() => {
-        try { // An exception on the listener should not break the SDK.
-          listener.logImpression({
-            impression,
-            attributes,
-            ip,
-            hostname,
-            sdkLanguageVersion
-          });
-        } catch (err) {
-          log.error(`Impression listener logImpression method threw: ${err}.`);
-        }
-      }, 0);
+      if (listener || integrationsManager) {
+        const impressionData = {
+          impression,
+          attributes,
+          ip,
+          hostname,
+          sdkLanguageVersion
+        };
```

Comment:
Be careful. While the data is the same, every time an object is passed to a customer to consume we should aim to sending a copy, not the reference.
If we pass along the reference and something unexpected is modified on it, it might impact other pieces of the SDK.

Reply:
I would send the same shallow-copied object (with a reference to impression) to both integrations and impressionListener, since it is the current way used for impressionListener, and because for avoiding any unexpected change we should do a deep copy of the object, that is expensive.